oshdb/src/test/java/org/heigit/ohsome/oshdb/contribution/ContributionsRelationTest.java

```diff
@@ -0,0 +1,315 @@
+package org.heigit.ohsome.oshdb.contribution;
+
+import static java.util.Collections.emptyList;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.CREATION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.DELETION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.GEOMETRY_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.MEMBER_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.ROLE_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.TAG_CHANGE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.EnumSet;
+import java.util.List;
+import org.heigit.ohsome.oshdb.OSHDBTest;
+import org.junit.jupiter.api.Test;
+
+class ContributionsRelationTest extends OSHDBTest {
+
+  @Test
+  void testMemberTagChanges() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(3, 3500, 253, 3, tags(), 1, 1),
+                node(2, 2500, 253, 3, tags(), 1, 1), // squash
+                node(1, 1100, 101, 1, tags(), 0, 0))), // minor-major change
+            way(1, 1000, 101, 1, tags(), mems(1))));
+
+    var versions = relations(1,
+        relation(2, 2000, 202, 2, tags(tag(1, 1)), mems(w(1, 0))),
+        relation(1, 1000, 101, 1, tags(), mems(w(1, 0), n(1, 0))));
+    var osh = osh(versions, nodes, ways);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(3500, 253, 3, versions.get(0), contrib);
+    assertEquals(EnumSet.of(GEOMETRY_CHANGE), contrib.getMinorTypes());
+
+    assertTrue(contribs.hasNext());
+    contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(TAG_CHANGE, MEMBER_CHANGE), contrib.getTypes());
```

Comment:
> Is it a problem to have the node 1 defined as member of the way differently than the node 1 defined as a member of the relation?
This was not an intention, that it is actually the same node, (i will change the node id in the way to 2)

> Why isn't this a GEOMETRY_CHANGE as well? Because the node with id = 1 is still used in the way that still is a member of the relation?
You are totally right, this MEMBER_CHANGE results in a GEOMETRY_CHANGE as well, so do most (but not all) MEMBER_CHANGEs. At the moment there is not an explicit check if the geometry really is different and I would like to see it more as a hint and the actually checking need to be done somewhere else.

> Do we need to check the resulting tags and members here?
Yes we could do.

Reply:
>> Do we need to check the resulting tags and members here?
> Yes we could do.

Actually, the tags, which are part of the entity, are checked within the `assertContrib` call.
- `assertEquals(entity, contrib.getEntity());`